functions/index.js

```diff
@@ -61,8 +71,33 @@ exports.newItemNotification = functions.region(region).firestore.document('items
 
     pushMessage(message, newItem.categoryString);
     return true;
-});
+}
+exports.chatNotificationCreate = functions.region(region).firestore.document('users/{userId}/chats/{chatId}/messages/{message}').onCreate((change, context) => onNewChat(change, context));
 
+/**
+ * Gets called every time a chat is added. It obtains the username of the sender from the database, 
+ * then sends a message with the author and the content of the message.
+ * @param change The data that has been updated/created
+ * @param context The context of the updated/created data, contains also the variables (wildcards) used in the document path.
+ * @returns true
+ */
+function onNewChat(change, context) {
+    const newChat = change.data();
+    db.collection("users").doc(newChat.from).get().then((user) => {
```

Comment:
```suggestion
    db.collection(DatabaseFields.DBFIELD_USERS).doc(newChat.from).get().then((user) => {
```

Reply:
There are defined constants in DatabaseFields, it's probably better to use them rather than strings (to avoid typos or debugging). I know that here everything is fine but it would just to make sure to use the same thing everywhere 😃

Reply:
I don't think I can access them from this javascript file